src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs

```diff
@@ -196,30 +196,24 @@ public string SetQuotedIdentifierString
             }
         }
 
-        public string SetNoExecString
+        public string GetSetNoExecString(bool on)
         {
-            get
-            {
-                return string.Format(System.Globalization.CultureInfo.InvariantCulture, s_SetNoExec, (this.settings.NoExec ? s_On : s_Off));
-            }
+            return string.Format(System.Globalization.CultureInfo.InvariantCulture, s_SetNoExec, (on ? s_On : s_Off));
         }
 
-        public string GetSetStatisticsTimeString(bool? on)
+        public string GetSetStatisticsTimeString(bool on)
         {
-            on = on ?? this.settings.StatisticsTime;
```

Comment:
the logic to set `on` to the value of the setting is incorrect since the setting is to determine if a SET option should be emitted, not the value of the SET option.  Fortunately, this fallback logic wasn't being used anywhere and can safely be removed.